### PR TITLE
feat(options): get cgroups container info

### PR DESCRIPTION
**Pull request by Qard**
Read container info from `/proc/self/cgroup` to get docker container id and kubernetes pod id default values.

### CODE REVIEW

**Comment by watson** [comment]
With #34, it was my intention that the 4 Kubernetes config option should be set automatically by the agent and not handled inside this repo. I'm open to moving this responsibility to this repo instead, but I'm not sure if that was what you intented?

Also, out of the 4 config options added in #34, this PR only sets `kubernetesPodUID`. Is that because that's the only variable that could be read from `/proc/self/cgroup`?

**Comment by watson** [comment]
> As for why I put it here, I figured by putting it here one could use the http-client without the agent and still have access to the container data.

Yeah, I guess it can be viewed from two different angles. I was thinking that this lib was low level and wouldn't try to "help" the user by filling these things out. But I can totally see your point. So I'm fine with doing it here.

File: index.js [hunk 1]
```diff
@@ -405,6 +408,15 @@ function normalizeOptions (opts) {
   // process
   normalized.serverUrl = parseUrl(normalized.serverUrl)
 
+  if (containerInfo) {
+    if (!normalized.containerId && containerInfo.containerId) {
+      normalized.containerId = containerInfo.containerId
+    }
+    if (!normalized.kubernetesPodUID && containerInfo.podId) {
+      normalized.kubernetesPodUID = containerInfo.podId
+    }
```
Comment (watson) [inline index.js]: Agreed on call: Add if block checking `!normalized.kubernetesPodName && containerInfo.podId` and if so set the pod name to the hostname

**Comment by watson** [comment]
It seems the tests needs to be updated with the new pod name.

And it would probably be a good idea if we updated the README where the kubernete config options are documented and describe that some of them will get populated automatically based on `/proc/self/cgroup`

**Comment by watson** [comment]
@Qard Did you see this comment?

> And it would probably be a good idea if we updated the README where the kubernete config options are documented and describe that some of them will get populated automatically based on `/proc/self/cgroup`

**Comment by Qard** [comment]
Yep, I'll get to that today. 👍

**Comment by watson** [comment]
Update: We'll just use the os hostname

**Pull request merged**